packages/deployments/contracts/contracts/core/connext/facets/TokenFacet.sol

```diff
@@ -211,7 +234,10 @@ contract TokenFacet is BaseConnextFacet {
     if (_canonical.domain == s.domain) {
       revert TokenFacet__setupAssetWithDeployedRepresentation_onCanonicalDomain();
     }
-    bytes32 key = _enrollAdoptedAndLocalAssets(_adoptedAssetId, _representation, _stableSwapPool, _canonical);
+    // Calculate the canonical key
+    bytes32 key = AssetLogic.calculateCanonicalHash(_canonical.id, _canonical.domain);
+    _enrollAdoptedAndLocalAssets(_adoptedAssetId, _representation, _stableSwapPool, _canonical, key);
+
```

Comment:
Isn't the cap supposed to be only on the canonical domain? (see line 241)
If so it shouldn't happen in this function  (also `_canonical.domain != s.domain` already checked in line 234)

Reply:
yep,  will remove the if block